### DISCUSSION TITLE
How to  auto activate menu Item ?

### DISCUSSION THREAD

**Discussion by prowong** [davetcc/tcMenu#346] [Q&A]
I made a volume controller that is controlled by tcmenu. I expect the volume item to be automatically activated after the menu times out, so that I can turn the rotary encoder directly to adjust the volume without pressing the OK button, so I added these codes, but the result is that it just navigates to the volume item without activating it. .Is there something I misunderstood?

`
renderer.setResetIntervalTimeSeconds(5);

renderer.setResetCallback([] {
        menuMgr.activateMenuItem(&menuVolume);
    });
`
I also tried menuMgr.setItemActive(&menuVolume) ,  but same result.

**Comment by davetcc** [comment]
Probably not the best way to do it but I _think_ it would work if you also did: `menuMgr.setCurrentEditor(&menuVolume);`

Most people wanting custom behaviour use a dashboard or take over the display, you can see there's a whole section of the documentation on taking over the display because it is such a common requirement - https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/renderer-take-over-display/

**Reply by prowong** [reply]
I added: menuMgr.setCurrentEditor(&menuVolume); item was activated, but the range of  values was not correct, it only had a maximum value from zero to the number of items. I also tried to add:  switches.pushSwitch(2,0); to simulate the key press, but nothing happen.

**Reply by davetcc** [reply]
I mean to be honest, nobody has tried to do this before, and it wasn't really considered in the use cases, I'll raise a bug to see if I can fix it in 4.0 but I think it would be far better to take over the display when the menu isn't needed but I'll try and make it work.

**Comment by davetcc** [comment]
See https://github.com/davetcc/tcMenu/issues/351
